packages/astro/src/vite-plugin-scripts/index.ts

```diff
@@ -0,0 +1,48 @@
+import { Plugin as VitePlugin } from 'vite';
+import { AstroConfig } from '../@types/astro.js';
+
+const SCRIPT_ID_PREFIX = `$scripts/`;
```

Comment:
I'd prefer if we followed [convention for virtual module](https://vitejs.dev/guide/api-plugin.html#virtual-modules-convention) ids. I could see potential for `$scripts` being a user-defined alias.

Reply:
Let me look into this more. I agree, but because we inject this as an import into our pages, the `\0` trick wasn't working when I tried it. I think I saw somewhere in Vite's codebase where they work around this by adding it back in the `resolveId()` hook, but then you're still saying that `$scripts/` is Astro-owned.

Maybe we take this opportunity to say that `$scripts` any any imports starting with `$` are invalid? Or, we use something less common, like `astro:scripts/page.js` or `$astro:scripts/page.js`

Reply:
updating to `astro:scripts/`. I don't think that the Vite docs on `virtual:` are accurate, there's no reference to them anywhere in the codebase. I think `astro:scripts/` does a good job of spiritually following this model

Reply:
Confirmed that Vite transforms/handles `\0` in some way that we can't match when we try to inject the import. Unclear if this is just some Vite magic that we need to duplicate, or if it's really impossible, but either way I think this is fine without the `\0` (even in the rollup ecosystem, it is optional, and I don't see SvelteKit using it).

Reply:
Works for me! Thanks for looking into it.